src/app/legacy/psammead/psammead-section-label/src/index.jsx

```diff
@@ -58,7 +67,11 @@ const SectionLabel = ({
   ...props
 }) => (
   <SectionLabelWrapper visuallyHidden={visuallyHidden} {...props}>
-    <Heading as={overrideHeadingAs}>
+    <Heading
+      as={overrideHeadingAs}
+      {...(labelId &&
+        !overrideHeadingAs && { id: `section-label-heading-${labelId}` })}
```

Comment:
Only apply the `id` for actual `h2` elements, as this is polymorphic component that can have its element type changed.